homeassistant/components/mill/climate.py

```diff
@@ -200,7 +207,7 @@ class LocalMillHeater(CoordinatorEntity, ClimateEntity):
     _attr_max_temp = MAX_TEMP
     _attr_min_temp = MIN_TEMP
     _attr_supported_features = ClimateEntityFeature.TARGET_TEMPERATURE
-    _attr_target_temperature_step = PRECISION_WHOLE
+    _attr_target_temperature_step = PRECISION_HALVES
```

Comment:
```suggestion
    _attr_target_temperature_step = PRECISION_HALVES if self._generation == 3 else PRECISION_WHOLE
```
If you do like this then I think it's enough to use the old code in set_temp but with float instead of int (as float without decimal will be returned as int)

Reply:
Looks good @gjohansson-ST ! Will test this out later (been on vacation for a while).

If I understands you correctly I can revert all the changes I did in `async_set_temperature`?

Reply:
Local API for Mill requires gen 3 heaters, and `PRECISION_HALVES` is always supported. So there is no need for `PRECISION_WHOLE` here.

Reply:
But tested this for the cloud integration (line 93), but I'm getting this error:
```NameError: name 'self' is not defined```